common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -552,7 +552,12 @@ def _render_student_view_for_items(self, context, display_items, fragment, view=
                 iteminfo['href'] = context.get('item_url', '').format(usage_key=usage_id)
             if is_user_authenticated:
                 if item.location.block_type == 'vertical' and completion_service:
-                    iteminfo['complete'] = completion_service.vertical_is_complete(item)
+                    try:
+                        is_complete = completion_service.vertical_is_complete(item)
+                    except AttributeError as error:
```

Comment:
If it's a generic catch all, shouldn't it be broader than AttributeError?

Reply:
Right, that would be the implementation _if_ we do want it generic.

I just wanted to catch the specific error first, then pose the question.

But yeah, I'd lean toward a broader error/exception.

Reply:
`vertical_is_complete` is also called in `vert_module.py` and potentially other places. Should we wrap those too? My hunch is that you're seeing it here because this is hit first, but I fear we might just push the exception down to the next layer. Although your changes in the completion repo might be alleviating this concern anyway, thought I'd mention

Reply:
@Dillon-Dumesnil The only non-test file I see any `grep` results for is this one. In fact, I don't see any file in `edx-platform` named `vert_module.py`. The closest I know of is `common/lib/xmodule/xmodule/vertical_block.py`, but it doesn't call this method.

Or am I missing something?

Reply:
Oh gah. Perhaps not. I was thinking of `vertical_block.py`, but I forgot that my team recently updated the logic to instead only check for assignments and not everything under a vertical. Thanks for double checking and I apologize for the miss!

Reply:
@Dillon-Dumesnil Cool, no worries! Just wanted to be sure I wasn't missing something!